src/Sulu/Bundle/WebsiteBundle/Twig/Seo/SeoTwigExtension.php

```diff
@@ -168,16 +168,22 @@ private function renderAlternateLinks(array $urls, $webspaceKey)
     {
         $html = '';
 
-        $portal = $this->requestAnalyzer->getPortal();
         $defaultLocale = null;
+        $portal = $this->requestAnalyzer->getPortal();
         if ($portal) {
-            $defaultLocale = $portal->getDefaultLocalization()->getLocalization();
-
-            $html .= $this->renderAlternateLink($urls[$defaultLocale], $webspaceKey, $defaultLocale, true);
+            $defaultLocale = $portal->getXDefaultLocalization()->getLocalization();
         }
 
         foreach ($urls as $locale => $url) {
-            $html .= $this->renderAlternateLink($url, $webspaceKey, $locale);
+            // url = '/' means that there is no translation for this page
+            // the only exception is the homepage where the requested resource-locator is false
+            if ($url !== '/' || $this->requestAnalyzer->getResourceLocator() === false) {
```

Comment:
hmm the problem here is, that this urls array is used in different usages. one for locale chooser where you want this URLs and links to homepage for locales which are not exists ...

and the other way is here where you dont want this links to homepage ...